eos/b-decays/b-to-psd-l-nu.cc

```diff
@@ -94,84 +93,90 @@ namespace eos
 
         std::shared_ptr<FormFactors<PToP>> form_factors;
 
-        // { U, q, I } -> { process, m_B, m_V, c_I }
-        // U: u, c; the quark flavor in the weak transition
+        // { q, P } -> { process, U, B_name, P_name, c_I }
         // q: u, d, s: the spectar quark flavor
-        // I: 1, 0, 1/2: the total isospin of the daughter meson
+        // P: D, K, pi: the type of daughter meson
         // process: string that can be used to obtain the form factor
-        // B: name of the B meson
-        // P: name of the daughter meson
+        // U: the quark flavor in the weak transition
+        // B_name: name of the B meson
+        // P_name: name of the daughter meson
         // c_I: isospin factor by which the amplitudes are multiplied
-        static const std::map<std::tuple<char, char, std::string>, std::tuple<std::string, std::string, std::string, double>> process_map;
+        static const std::map<std::tuple<char, std::string>, std::tuple<std::string, char, std::string, std::string, double>> process_map;
 
         inline std::string _process() const
         {
-            const char U = opt_U.value()[0];
             const char q = opt_q.value()[0];
-            const std::string I = opt_I.value();
-            const auto p = process_map.find(std::make_tuple(U, q, I));
+            const std::string P = opt_P.value();
+            const auto p = process_map.find(std::make_tuple(q, P));
 
             if (p == process_map.end())
-                throw InternalError("Unsupported combination of U=" + stringify(U) + ", q=" + q + ", I=" + I);
+                throw InternalError("Unsupported combination of q = " + stringify(q) + ", P = " + P);
 
             return std::get<0>(p->second);
         }
 
-        inline std::string _B() const
+        inline char _U() const
```

Comment:
Can we turn this from a ``char`` to a ``QuarkFlavor``? Then in the below we can turn the sequence of if statements into a ``switch`` statement.